translate/storage/properties.py

```diff
@@ -285,7 +285,7 @@ def encode(string: str, encoding: str | None = None) -> str:
         # FIXME: dialects are a bad idea, not possible for subclasses
         # to override key methods
         if encoding not in {"utf-8", "utf-16"}:
-            return javapropertiesencode(string or "")
+            return javapropertiesencode(string or "", encoding or "iso-8859-1")
```

Comment:
I think passing `None` here and dealing with it in `javapropertiesencode` is better approach.